data_steward/test/unit_test/vocabulary_test.py

```diff
@@ -0,0 +1,97 @@
+import unittest
+import StringIO
+from vocabulary import *
+from vocabulary import _transform_csv
+import test_util
+import os
+import mock
+
+
+class VocabularyTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        super(VocabularyTest, self).setUp()
+
+    @staticmethod
+    def do_transform_file(input_text):
+        in_fp = StringIO.StringIO(input_text)
+        out_fp = StringIO.StringIO()
+        _transform_csv(in_fp, out_fp)
+        return out_fp.getvalue()
+
+    def test_transform_file(self):
+        header = ['concept_id', 'valid_start_date', 'valid_end_date', 'invalid_reason']
+        row1 = ['1', '2017-05-17', '2099-12-31', '']
+        row2 = ['2', '2019-01-01', '2099-12-31', '']
+        rows = [header, row1, row2]
+        lines = map(lambda r: DELIMITER.join(r), rows)
+
+        # after transform should output valid csv with date format yyyy-mm-dd
+        expected_text = LINE_TERMINATOR.join(lines) + LINE_TERMINATOR
+        input_text = expected_text.replace('-', '')
+        actual_text = VocabularyTest.do_transform_file(input_text)
+        self.assertEqual(expected_text, actual_text)
+
+        # windows line endings are replaced with linux ones
+        input_text = expected_text.replace(LINE_TERMINATOR, '\r\n')
+        actual_text = VocabularyTest.do_transform_file(input_text)
+        self.assertEqual(expected_text, actual_text)
```

Comment:
Because you are testing two functionalities in the same function, I would suggest adding a message in case one of your assertions fails.  This will help make it clearer to the maintainer what has changed if this test fails.  

`self.assertEqual(expected_text, actual_text, "The date formatted text did not format as expected.")`
and
`self.assertEqual(expected_text, actual_text, "The Windows line endings were not replaced as expected.")`